core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -1733,22 +1706,19 @@ public void testRemoveIdentifierFields() {
             .setIdentifierFields("new_field", "new_field2")
             .apply();
 
-    Assert.assertEquals(
-        "remove an identifier field should succeed",
-        Sets.newHashSet(
-            newSchema.findField("new_field").fieldId(),
-            newSchema.findField("new_field2").fieldId()),
-        newSchema.identifierFieldIds());
+    assertThat(newSchema.identifierFieldIds())
+        .as("remove an identifier field should succeed")
+        .isEqualTo(
```

Comment:
it's fine to also use `containsExactly()` here (and in other places)

Reply:
I will include the changes here in next PR.